### PR TITLE
Added Self Heal exporter

### PR DESCRIPTION
**Pull request by vredara**
Signed-off-by: venkata edara <redara@redhat.com>

### CODE REVIEW

**Comment by vredara** [comment]
@aravindavk @shtripat please review this PR which exposed gluster heal cnt using glusterutils interface

File: gluster-exporter/metric_volume.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			Name: "volume",
+			Help: "Volume Name",
+		},
+		{
+			Name: "brick_path",
+			Help: "Brick Path",
+		},
+		{
+			Name: "host",
+			Help: "Hostname",
```
Comment (shtripat) [inline gluster-exporter/metric_volume.go]: Host FQDN or IP?

File: gluster-exporter/metric_volume.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+}
+
+func healCounts() error {
+	isLeader, err := gluster.IsLeader()
+	if !isLeader || err != nil {
+		// Unable to find out if the current node is leader
+		// Cannot register volume metrics at this node
+		log.WithError(err).Error("Unable to find if the current node is leader")
```
Comment (shtripat) [inline gluster-exporter/metric_volume.go]: It will good to add hostname as well in the log I feel

File: gluster-exporter/metric_volume.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		return err
+	}
+
+	for _, volume := range volumes {
+		name := volume.Name
+		heals, err := gluster.HealInfo(name)
+		if err != nil {
+			log.WithError(err).WithFields(log.Fields{
+				"volume": name,
+			}).Error("Error getting disk usage")
```
Comment (shtripat) [inline gluster-exporter/metric_volume.go]: disk usage or heal info?

File: gluster-exporter/metric_volume.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		},
+		{
+			Name: "host",
+			Help: "Hostname",
+		},
+	}
+
+	glusterVolumeHealCount = newPrometheusGaugeVec(Metric{
+		Namespace: "gluster",
+		Name:      "volume_heal_count",
```
Comment (shtripat) [inline gluster-exporter/metric_volume.go]: Is this heal pending count?

Comment (vredara) [reply gluster-exporter/metric_volume.go]: yes. pending heal cnt.

File: gluster-exporter/metric_volume.go (hunk 3 above)
Comment (vredara) [reply gluster-exporter/metric_volume.go]: it is typo, Yes it is heal info. I will fix this.

File: gluster-exporter/metric_volume.go (hunk 1 above)
Comment (vredara) [reply gluster-exporter/metric_volume.go]: IP addr. should I change the label here?.

Comment (shtripat) [reply gluster-exporter/metric_volume.go]: All other places we saying something like 

```
                {
                        Name: "host",
                        Help: "Host name or IP",
                },
```

File: gluster-exporter/metric_volume.go (hunk 4 above)
Comment (shtripat) [reply gluster-exporter/metric_volume.go]: Ok. In that case should we rename the metric as `volume_pending_heal_count` ?

File: gluster-exporter/metric_volume.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		"volume":     volname,
+		"brick_path": brick,
+		"host":       host,
+	}
+
+}
+
+func healCounts() error {
+	isLeader, err := gluster.IsLeader()
+	if !isLeader || err != nil {
```
Comment (aravindavk) [inline gluster-exporter/metric_volume.go]: This message will be logged if current exporter is not leader.

```
if err != nil {
    // Log
    return err
}
if !isLeader {
    return nil
}
```

Comment (vredara) [reply gluster-exporter/metric_volume.go]: ack. will fix this

File: gluster-exporter/metric_volume.go (hunk 4 above)
Comment (shtripat) [reply gluster-exporter/metric_volume.go]: ok

**Comment by shtripat** [comment]
@vredara add an entry into config file as well to enable this.

**Comment by vredara** [comment]
@shtripat @aravindavk pls review this. I added the heal entry into metric doc and also conf file.

**Pull request merged**